.github/workflows/release.yml

```diff
@@ -0,0 +1,112 @@
+# This automation is to perform release tagging and artifact uploads of the LFS objects
+
+name: Publish new release
+
+on:
+  # Allows you to run this workflow manually from the Actions tag
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: 'The repo to target for release'
+        required: true
+        type: choice
+        default: o3de
+        options:
+          - o3de
+          - o3de-atom-sampleviewer
+          - o3de-multiplayersample
+          - o3de-netsoaktest
+      release_tag:
+        description: 'The release tag'
+        required: true 
+        type: string
+      release_notes_url:
+        description: 'The release notes url'
+        required: false
+        type: string
+      draft_release:
+        description: 'Stage release as draft. If unchecked, publish immediately'
+        default: true
+        required: true
+        type: boolean
+        
+run-name: Releasing ${{ inputs.repository }} for ${{ inputs.release_tag }}
+
+jobs:
+  Publish:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: ${{ github.ref_name }} # Check environment against branch
+    
+    permissions:
+      contents: write
+    
+    steps:
+      # Checks-out the repository under $GITHUB_WORKSPACE, so the job can access it
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository_owner }}/${{ inputs.repository }}
+          ref: main
+          fetch-depth: 0
+      
+      - name: Git LFS pull
+        run: |
+          git lfs install
+          git lfs fetch
+      
+      - name: Get repo vars
+        id: vars
+        run: |
+          echo ::set-output name=latest_tag::$(git describe --tags `git rev-list --tags --max-count=1`)
+          echo ::set-output name=latest_main_sha::$(git log -n 1 --pretty=format:"%H" origin/main)
+          
+      - name: Archive repo as compressed files
+        run: |
+          git archive -o ${{ inputs.repository }}-${{ inputs.release_tag }}-lfs.zip HEAD
+          git archive -o ${{ inputs.repository }}-${{ inputs.release_tag }}-lfs.tar.gz HEAD
+          
+      # Creates a release tag based on imputs
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@latest
```

Comment:
Do you think it's an issue that the projects for the `create-release` and `upload-release-asset` actions are now archived? I don't know how often the API for github releases change. 

I do see an up-to-date version of this action that performs the tag creation and upload here: https://github.com/ncipollo/release-action

Reply:
Currently, it's not a huge issue, the APIs are generally backwards compatible to prevent large scale impact. There is some concern that the other actions being called out are not as secure as the official one.

That said, the cost to switch over to another action is not particularly huge, since the attributes for release are generally the same. I'm totally fine with moving to a more frequently updated action for maintainability.

Reply:
Got it. I'm fine with using the actions defined for now. After this release we can either find a better supported one or fork the deprecated repo update it ourselves.

Reply:
Ended up using the one you suggested, as it does parallel uploads and it's very actively maintained